app/views/layouts/previews.html.erb

```diff
@@ -0,0 +1,26 @@
+<% content_for :view_component_preview_source do %>
+  <link href="<%= asset_path('prism.css', skip_pipeline: true) %>" media="screen" rel="stylesheet" type="text/css" />
+  <div class="view-component-source-example">
+    <h2>Source:</h2>
+    <pre class="source">
+      <% if @render_args[:component] %>
+        <code class="language-ruby">
+          <%=h @preview.preview_example_source(@example_name) %>
+        </code>
+      <% else %>
+        <% template = @view_renderer.lookup_context.find_template(@render_args[:template]) %>
+        <code class="language-<%= prism_language_name(template: template) %>">
+          <%=h template.source %>
+        </code>
+      <% end %>
+    </pre>
+  </div>
+  <script type="text/javascript" src="<%= asset_path('prism.min.js', skip_pipeline: true) %>"></script>
+<% end %>
+
+<% if @host_application_layout == false %>
+  <%= yield %>
+  <%= yield(:view_component_preview_source) %>
```

Comment:
Can you share what's happening here?

Reply:
`yield` on line 22 renders the view component, `yield(:view_component_preview_source)` on line 23 renders source code example.

Reply:
Wouldn't you expect to guard this with `if content_for?(:view_component_preview_source) %>` ?